fe/fe-core/src/main/java/com/starrocks/sql/ast/ColumnDef.java

```diff
@@ -266,9 +267,28 @@ public Type getType() {
         return typeDef.getType();
     }
 
+    public void analyze(boolean isOlap, boolean isInternalCatalog, String engineName) throws AnalysisException {
+        if (isInternalCatalog) {
```

Comment:
could we create not null column for exteranl catalog?

Reply:
this pr only focus on internalCatalog create external table, not for external catalog create table such as iceberg sink